src/public/modules/forms/admin/controllers/edit-fields-modal.client.controller.js

```diff
@@ -74,13 +74,23 @@ function EditFieldsModalController(
   vm.user = Auth.getUser() || $state.go('signin')
   if (vm.field.fieldType === 'email') {
     const userEmailDomain = '@' + vm.user.email.split('@').pop()
+
+    // Backwards compatibility and inconsistency fix.
+    // Set allowedEmailDomains array to empty if allow domains toggle is off.
+    if (vm.field.hasAllowedEmailDomains === false) {
+      vm.field.allowedEmailDomains = []
+    } else {
+      // hasAllowedEmailDomains is true, set "true" state based on length of allowedEmailDomains.
+      vm.field.hasAllowedEmailDomains = vm.field.allowedEmailDomains.length > 0
+    }
+
     vm.field.allowedEmailDomainsPlaceholder = `${userEmailDomain}\n@agency.gov.sg`
-    if (vm.field.allowedEmailDomains.length > 0) {
+    if (vm.field.hasAllowedEmailDomains) {
```

Comment:
if the end goal is to remove `hasAllowedEmailDomains`, why not just stick to using `allowedEmailDomains.length`?

Reply:
Case 2 in the table above would incorrectly be enabled

Reply:
Oh but you are right, we have already reset the array before his

Reply:
Clarified irl: hasAllowedEmailDomains is still needed in the client to show the state of the "Restrict email domains" toggle.